core/src/main/java/org/infinispan/commands/control/LockControlCommand.java

```diff
@@ -149,7 +148,7 @@ public void setParameters(int commandId, Object[] args) {
       globalTx = (GlobalTransaction) args[i++];
       unlock = (Boolean) args[i++];
       keys = (List<Object>) args[i++];
-      flags = (Set<Flag>) args[i];
+      flags = (EnumSet<Flag>) args[i];
```

Comment:
Ah yes, I wanted to convert everything to an `EnumSet` in `setFlags()`, but then I backed down on it. Reverting.